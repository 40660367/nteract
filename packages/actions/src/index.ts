/**
 * @module actions
 */

import * as ActionTypes from "./actionTypes/index";

console.log("actiontypes in actions/actiontypes", ActionTypes);

export * from "./actions/index";
export * from "./actionTypes/index";
