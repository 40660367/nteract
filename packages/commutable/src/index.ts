/**
 * @module commutable
 */
// .....................................
// API Exports
//

export * from "./primitives";
export * from "./structures";
export * from "./outputs";
export * from "./cells";
export * from "./notebook";

/*
// from structures
export {
  emptyCodeCell,
  emptyMarkdownCell,
  emptyNotebook,
  monocellNotebook,
  createCodeCell,
  insertCellAt,
  insertCellAfter,
  deleteCell,
  appendCellToNotebook
} from "./structures";

// v4
export { StreamOutput, Output, createImmutableOutput } from "./v4";

export {
  createImmutableMimeBundle,
  makeDisplayData,
  makeErrorOutput,
  makeStreamOutput,
  makeExecuteResult,
  MimeBundle
} from "./outputs";

export {
  makeRawCell,
  makeCodeCell,
  makeMarkdownCell,
  ImmutableCodeCell,
  ImmutableMarkdownCell,
  ImmutableRawCell,
  ImmutableCell,
  CellType
} from "./cells";

export {
  toJS,
  stringifyNotebook,
  fromJS,
  parseNotebook,
  makeNotebookRecord,
  Notebook,
  ImmutableNotebook
} from "./notebook";

*/
