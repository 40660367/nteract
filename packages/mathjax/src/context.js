/* @flow strict */
import * as React from "react";

export type MathJaxObject = {
  Hub: {
    getJaxFor: HTMLElement => void,
    Config: (options: *) => void,
    Register: {
      StartupHook: (str: string, cb: () => void) => void,
      MessageHook: (string, cb: (msg: string) => void) => void
    },
    Reprocess: (HTMLElement, cb: *) => *,
    Queue: (*) => void,
    processSectionDelay: number
  }
};

export type MathJaxContextValue = {
  MathJax: ?MathJaxObject,
  input: "tex" | "ascii"
};

const MathJaxContext: React.Context<MathJaxContextValue> = React.createContext({
  MathJax: null,
  input: "tex"
});

export default MathJaxContext;
