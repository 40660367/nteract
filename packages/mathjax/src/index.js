// @flow

import Node from "./node";
import Context from "./context";
import Text from "./text";

export { Node, Context, Text };
export default { Node, Context, Text };
