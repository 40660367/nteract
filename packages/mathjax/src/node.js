// @flow
import * as React from "react";
import PropTypes from "prop-types";

const types = {
  ascii: "asciimath",
  tex: "tex"
};

import MathJaxContext, {
  type MathJaxObject,
  type MathJaxContextValue
} from "./context";

type Props = {
  inline: boolean,
  children: string,
  onRender: ?Function
};

class MathJaxNode_ extends React.Component<Props & MathJaxContextValue, null> {
  script: ?HTMLScriptElement;
  nodeRef: React.ElementRef<*>;

  static defaultProps = {
    inline: false,
    onRender: null
  };

  constructor(props: Props & MathJaxContextValue) {
    super(props);

    this.nodeRef = React.createRef();

    (this: any).typeset = this.typeset;
  }

  /**
   * Render the math once the node is mounted
   */
  componentDidMount() {
    this.typeset();
  }

  /**
   * Update the jax, force update if the display mode changed
   */
  componentDidUpdate(prevProps: Props & MathJaxContextValue) {
    const forceUpdate =
      prevProps.inline !== this.props.inline ||
      prevProps.children !== this.props.children;
    this.typeset(forceUpdate);
  }

  /**
   * Clear the math when unmounting the node
   */
  componentWillUnmount() {
    this.clear();
  }

  /**
   * Clear the jax
   */
  clear() {
    const MathJax = this.props.MathJax;

    if (!MathJax) {
      return;
    }

    if (!this.script) {
      return;
    }

    const jax = MathJax.Hub.getJaxFor(this.script);

    if (jax) {
      jax.Remove();
    }
  }

  /**
   * Update math in the node
   * @param { Boolean } forceUpdate
   */
  typeset(forceUpdate: boolean = false) {
    const { MathJax } = this.props;

    if (!MathJax || !MathJax.Hub) {
      throw Error(
        "Could not find MathJax while attempting typeset! It's likely the MathJax script hasn't been loaded or MathJax.Context is not in the hierarchy"
      );
    }

    const text = this.props.children;

    if (forceUpdate) {
      this.clear();
    }

    if (forceUpdate || !this.script) {
      this.setScriptText(text);
    }
    // As an invariant of above, this shouldn't occur
    if (!this.script) {
      return;
    }

    MathJax.Hub.Queue(MathJax.Hub.Reprocess(this.script, this.props.onRender));
  }

  /**
   * Create a script
   * @param { String } text
   */
  setScriptText(text: *) {
    const inline = this.props.inline;
    const type = types[this.props.input];
    if (!this.script) {
      this.script = document.createElement("script");
      this.script.type = `math/${type}; ${inline ? "" : "mode=display"}`;

      this.nodeRef.current.appendChild(this.script);
    }

    // It _should_ be defined at this point, we'll just return at this point now
    if (!this.script) {
      return;
    }

    if ("text" in this.script) {
      // IE8, etc
      this.script.text = text;
    } else {
      this.script.textContent = text;
    }
  }

  render() {
    return <span ref={this.nodeRef} />;
  }
}

class MathJaxNode extends React.PureComponent<Props, null> {
  render() {
    return (
      <MathJaxContext.Consumer>
        {({ MathJax, input }: MathJaxContextValue) => {
          if (!MathJax) {
            return null;
          }

          return (
            <MathJaxNode_ {...this.props} input={input} MathJax={MathJax} />
          );
        }}
      </MathJaxContext.Consumer>
    );
  }
}

export default MathJaxNode;
