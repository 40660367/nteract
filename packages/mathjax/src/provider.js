// @flow strict
/* global MathJax */

import * as React from "react";
import loadScript from "./load-script";

import MathJaxContext, {
  type MathJaxObject,
  type MathJaxContextValue
} from "./context.js";

// MathJax expected to be a global and may be undefined
declare var MathJax: ?MathJaxObject;

declare type onLoad = () => void;
type Props = {
  src: ?string,
  children: React.Node,
  didFinishTypeset: ?() => void,
  // Provide a way to override how we load MathJax and callback to the onLoad
  // For Hydrogen, for instance we can set
  //
  //  loader={(onLoad) => loadMathJax(document, onLoad)}
  //
  onLoad: ?onLoad,
  loader: ?(cb: onLoad) => void,
  input: "ascii" | "tex",
  delay: number,
  options: ?*,
  loading: React.Node,
  noGate: boolean,
  onError: (err: Error) => void
};

type State = MathJaxContextValue;

/**
 * MathJax Provider
 */
class Provider extends React.Component<Props, State> {
  static defaultProps = {
    src:
      "https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.1/MathJax.js?config=TeX-MML-AM_CHTML",
    input: "tex",
    didFinishTypeset: null,
    loader: null,
    delay: 0,
    options: {},
    loading: null,
    noGate: false,
    onLoad: null,
    onError: (err: Error) => {
      console.error(err);
    }
  };

  constructor(props: Props) {
    super(props);

    this.state = {
      MathJax: undefined,
      // TODO: Ensure state gets updated when the input prop changes
      input: this.props.input,
      hasProviderAbove: true
    };
  }

  componentDidMount() {
    const src = this.props.src;

    if (src == null) {
      return this.onLoad();
    }

    if (!this.props.loader) {
      loadScript(src, this.onLoad);
    } else {
      this.props.loader(this.onLoad);
    }
  }

  onLoad = () => {
    if (!MathJax || !MathJax.Hub) {
      this.props.onError(
        new Error("MathJax not really loaded even though onLoad called")
      );
      return;
    }

    const options = this.props.options;

    MathJax.Hub.Config(options);

    MathJax.Hub.Register.StartupHook("End", () => {
      if (!MathJax) {
        this.props.onError(
          new Error("MathJax became undefined in the middle of processing")
        );
        return;
      }
      MathJax.Hub.processSectionDelay = this.props.delay;

      if (this.props.didFinishTypeset) {
        this.props.didFinishTypeset();
      }
    });

    MathJax.Hub.Register.MessageHook("Math Processing Error", message => {
      if (this.props.onError) {
        this.props.onError(new Error(message));
      }
    });

    if (this.props.onLoad) {
      this.props.onLoad();
    }

    this.setState({
      MathJax
    });
  };

  render() {
    return (
      <MathJaxContext.Provider value={this.state}>
        {this.props.children}
      </MathJaxContext.Provider>
    );
  }
}

export default Provider;
