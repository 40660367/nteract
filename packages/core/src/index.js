//@flow
import * as actions from "./actions";
import * as actionTypes from "./actionTypes";
import * as middlewares from "./middlewares";
import * as reducers from "./reducers";
import * as themes from "./themes";
import * as epics from "./epics";
import * as selectors from "./selectors";
import * as records from "./records";

export * from "./components";
export * from "./providers";

// keeping with backwards compatiblity for now
const constants = actionTypes;

export {
  records,
  actions,
  actionTypes,
  constants,
  middlewares,
  reducers,
  themes,
  selectors,
  epics
};
