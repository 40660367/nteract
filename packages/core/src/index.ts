import * as reducers from "@nteract/reducers";
import * as epics from "@nteract/epics";
import * as selectors from "@nteract/selectors";
import * as state from "@nteract/types";
import * as actions from "@nteract/actions";

import * as middlewares from "./middlewares";

export * from "@nteract/types";

console.log("actions in core import", actions);

export { actions, middlewares, reducers, selectors, epics, state };
