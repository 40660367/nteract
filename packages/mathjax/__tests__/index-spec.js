// @flow

import * as React from "react";
import { Node, Provider } from "../src";
import renderer from "react-test-renderer";

import toJson from "enzyme-to-json";

import Enzyme, { shallow, mount } from "enzyme";
import Adapter from "enzyme-adapter-react-16";

describe("MathJax", () => {
  test("Node can be renderered without provider", () => {
    const wrapper = mount(<Node>x^2 + y = 3</Node>);
    expect(toJson(wrapper)).toMatchSnapshot(name);
  });

  test("Nodes use existing provider", () => {
    const wrapper = mount(
      <Provider>
        <Node>x^2 + y = 3</Node>
        <Node>x^3 + y = 2</Node>
      </Provider>
    );
    expect(toJson(wrapper)).toMatchSnapshot(name);
  });
});
