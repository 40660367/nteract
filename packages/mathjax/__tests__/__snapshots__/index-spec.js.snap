// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MathJax Node can be renderered without provider: nodejs 1`] = `
<MathJaxNode
  inline={false}
  onRender={null}
>
  <Provider
    delay={0}
    didFinishTypeset={null}
    input="tex"
    loading={null}
    noGate={false}
    onError={[Function]}
    onLoad={null}
    options={Object {}}
    src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.1/MathJax.js?config=TeX-MML-AM_CHTML"
  >
    <MathJaxNode
      inline={false}
      onRender={null}
    />
  </Provider>
</MathJaxNode>
`;

exports[`MathJax Nodes use existing provider: nodejs 1`] = `
<Provider
  delay={0}
  didFinishTypeset={null}
  input="tex"
  loading={null}
  noGate={false}
  onError={[Function]}
  onLoad={null}
  options={Object {}}
  src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.1/MathJax.js?config=TeX-MML-AM_CHTML"
>
  <MathJaxNode
    inline={false}
    onRender={null}
  />
  <MathJaxNode
    inline={false}
    onRender={null}
  />
</Provider>
`;
